unit_test/CMakeLists.txt

```diff
@@ -34,6 +37,47 @@ TARGET_INCLUDE_DIRECTORIES(kokkoskernels_gtest PUBLIC $<BUILD_INTERFACE:${GTEST_
 #Gtest minimally requires C++11
 TARGET_COMPILE_FEATURES(kokkoskernels_gtest PUBLIC cxx_std_11)
 
+
+
+#
+#
+#  Making some CMake experiments to design a more automated
+#  unit-test framework that generates automatically cpp files
+#  needed by each backend to generate the unit-test executables.
+#
+#
+
+set(MY_COMPS Common)
+set(MY_BACKENDS Serial OpenMP)
+
+foreach(COMP ${MY_COMPS})
+  string(TOLOWER ${COMP} COMP_LOWER)
+  file(GLOB COMP_CPP ${COMP_LOWER}/*.hpp)
+  message(STATUS "List of Test_${COMP}_.cpp files:")
+  foreach(BACKEND ${MY_BACKENDS})
+    string(TOLOWER ${BACKEND} BACKEND_LOWER)
+    string(TOUPPER ${BACKEND} BACKEND_UPPER)
+    if(KOKKOS_ENABLE_${BACKEND_UPPER})
+      foreach(COMP_FILE ${COMP_CPP})
+        get_filename_component(COMP_NAME ${COMP_FILE} NAME_WE)
+        string(REPLACE "Test_${COMP}_" "" COMP_TEST ${COMP_NAME})
+        message(STATUS "  ${COMP_TEST}")
+        string(COMPARE EQUAL ${COMP_TEST} float128 CHECK_FLOAT)
+        if(${CHECK_FLOAT})
```

Comment:
You can do this as:
````cmake
if ("${COMP_TEST}" STREQUAL float128)
...
````

Reply:
Yes, I have been very very careful to perform each step explicitly to help with debugging, at some point I was printing every single variable I created x)
I will make things more compact as suggested.